src/utils/issue.ts

```diff
@@ -0,0 +1,171 @@
+import { Context } from "../types/context";
+import { Issue, ISSUE_TYPE } from "../types/payload";
+import { getLinkedPullRequests } from "./get-linked-prs";
+
+export function isParentIssue(body: string) {
+  const parentPattern = /-\s+\[( |x)\]\s+#\d+/;
+  return body.match(parentPattern);
+}
+
+export async function getAssignedIssues(context: Context, username: string): Promise<Issue[]> {
+  const payload = context.payload;
+
+  try {
+    return await context.octokit.paginate(
+      context.octokit.issues.listForRepo,
+      {
+        owner: payload.repository.owner.login,
+        repo: payload.repository.name,
+        state: ISSUE_TYPE.OPEN,
+        per_page: 1000,
+      },
+      ({ data: issues }) => issues.filter((issue) => !issue.pull_request && issue.assignee && issue.assignee.login === username)
+    );
+  } catch (err: unknown) {
+    context.logger.fatal("Fetching assigned issues failed!", err);
+    return [];
+  }
+}
+
+export async function addCommentToIssue(context: Context, message: string | null) {
+  let comment = message as string;
+
+  const { payload } = context;
+
+  const issueNumber = payload.issue.number;
+  try {
+    await context.octokit.issues.createComment({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: issueNumber,
+      body: comment,
+    });
+  } catch (e: unknown) {
+    context.logger.fatal("Adding a comment failed!", e);
+  }
+}
+
+//// Pull Requests \\\\
+
+export async function closePullRequest(context: Context, pullNumber: number) {
+  const { repository } = context.payload;
+  try {
+    await context.octokit.rest.pulls.update({
+      owner: repository.owner.login,
+      repo: repository.name,
+      pull_number: pullNumber,
+      state: "closed",
+    });
+  } catch (err: unknown) {
+    context.logger.fatal("Closing pull requests failed!", err);
+  }
+}
+
+export async function closePullRequestForAnIssue(context: Context, issueNumber: number, repository: Context["payload"]["repository"]) {
+  const logger = context.logger;
+  if (!issueNumber) {
+    throw logger.fatal("Issue is not defined");
+  }
+
+  const linkedPullRequests = await getLinkedPullRequests(context, {
+    owner: repository.owner.login,
+    repository: repository.name,
+    issue: issueNumber,
+  });
+
+  if (!linkedPullRequests.length) {
+    return logger.info(`No linked pull requests to close`);
+  }
+
+  logger.info(`Opened prs`, linkedPullRequests);
+  let comment = `These linked pull requests are closed: `;
+  for (let i = 0; i < linkedPullRequests.length; i++) {
+    await closePullRequest(context, linkedPullRequests[i].number);
+    comment += ` <a href="${linkedPullRequests[i].href}">#${linkedPullRequests[i].number}</a> `;
```

Comment:
```suggestion
    comment += ` ${linkedPullRequests[i].href} `;
```

Not sure if this is better, but I remember seeing this render in a diff block so there's no need to dress it up if formatting isn't supported.